src/Symfony/Component/VarDumper/Cloner/VarCloner.php

```diff
@@ -129,39 +129,44 @@ protected function doClone($var)
                             continue 2;
                         }
                         $stub = $arrayStub;
+
+                        if (\PHP_VERSION_ID >= 80100) {
+                            $stub->class = array_is_list($v) ? Stub::ARRAY_INDEXED : Stub::ARRAY_ASSOC;
+                            $a = $v;
+                            break;
+                        }
+
                         $stub->class = Stub::ARRAY_INDEXED;
 
                         $j = -1;
                         foreach ($v as $gk => $gv) {
```

Comment:
shouldn't this loop in the a `else` block ?

Reply:
I'm not sure what you mean, there's a `break` in the `if` above

Reply:
indeed. I looked at it too fast